doc/source/data/data-internals.rst

```diff
@@ -11,79 +11,164 @@ For a gentler introduction to Ray Data, see :ref:`Quickstart <data_quickstart>`.
 
 .. _dataset_concept:
 
+Key concepts
+============
+
 Datasets and blocks
-===================
+-------------------
 
-A :class:`Dataset <ray.data.Dataset>` operates over a sequence of Ray object references
-to :term:`blocks <Block>`. Each block contains a disjoint subset of rows, and Ray Data
-loads and transforms these blocks in parallel.
+Datasets
+~~~~~~~~
 
-The following figure visualizes a dataset with three blocks, each holding 1000 rows.
-The :class:`~ray.data.Dataset` is the user-facing Python object
-(usually held in the driver), while materialized blocks are stored as objects in Ray's
-shared-memory :ref:`object store <objects-in-ray>`.
+:class:`Dataset <ray.data.Dataset>` is the main user-facing Python API. It represents a 
+distributed data collection, and defines data loading and processing operations. You 
+typically use the API in this way:
+
+1. Create a Ray Dataset from external storage or in-memory data.
+2. Apply transformations to the data. 
+3. Write the outputs to external storage or feed the outputs to training workers. 
+
+Blocks
+~~~~~~
 
-Generally speaking, the number of concurrently executing tasks determines CPU utilization (or GPU utilization if using GPU transforms).
-The total heap memory usage is a function (determined by your UDF) of the number of concurrently executing tasks multiplied by the block size.
-The total number of materialized blocks in scope determines Ray's object store usage; if this exceeds Ray's object store capacity, then Ray automatically spills blocks to disk.
-Ray Data uses :ref:`streaming execution <streaming_execution>` to minimize the total number of materialized blocks in scope and therefore avoid spilling.
+A *block* is the basic unit of data bulk that Ray Data stores in the object store and 
+transfers over the network. Each block contains a disjoint subset of rows, and Ray Data 
+loads and transforms these blocks in parallel. 
+
+The following figure visualizes a dataset with three blocks, each holding 1000 rows.
+Ray Data holds the :class:`~ray.data.Dataset` on the driver and stores the materialized 
```

Comment:
* `on the driver` -> `on the process that triggers execution, which is usually the driver.`
* `stores the materialized blocks`: remove `materialized`? it sounds like we only store the results of "materialize` in the object store.